tests/unit_tests/test_array_element.py

```diff
@@ -85,10 +85,18 @@ def test_array_element(array_element_process_code, example_input, expected_outpu
     ],
 )
 def test_array_element_exceptions(array_element_process_code, example_input, raises_exception, error_message):
-    if raises_exception:
-        with pytest.raises(Exception) as exc:
-            run_process(array_element_process_code, "array_element", example_input)
-        assert error_message in str(exc.value)
-
-    else:
-        run_process(array_element_process_code, "array_element", example_input)
+    additional_js_code_to_run = (
+        load_datacube_code()
+        + f"const d = {json.dumps(example_input['data']) if 'data' in example_input else 'undefined'};"
+        + f"d.labels = {json.dumps(example_input['labels']) if 'labels' in example_input else 'undefined'};"
+        if "data" in example_input
+        else ""
+    )
+    process_arguments = f"{{...{json.dumps(example_input)}, 'data': {'d' if 'data' in example_input else 'undefined'}}}"
```

Comment:
I believe that this `'d'` will be interpreted as a string and not as a variable `d` when JS code will be run. I am not actually 100% sure about this, if I am wrong then fftr, just wanted to point this out and make sure everything is ok.

Reply:
Well, one can always test it out. For example, we can put the following in the python console:
```python
example_input = {"data": 1}
print(f"{{...{json.dumps(example_input)}, 'data': {'d' if 'data' in example_input else 'undefined'}}}")
>> {...{"data": 1}, 'data': d}
```
There are no quotes around `d`, so it will be interpreted as a variable. Let's check that in JS console:
```js
{...{"data": 1}, 'data': d}
>> Uncaught ReferenceError: d is not defined
```
Similar for undefined:
```python
example_input = {}
print(f"{{...{json.dumps(example_input)}, 'data': {'d' if 'data' in example_input else 'undefined'}}}")
>> {...{}, 'data': undefined}
```
In JS:
```js
{...{}, 'data': undefined}
>> {data: undefined}
```
-----------------------------
There used to be a bug in line 47 of `test_array_labels` though. I changed the line:
```python
 f"const d = {json.dumps(example_input['data'] if 'data' in example_input else 'undefined')};"
 ```
to
```python
f"const d = {json.dumps(example_input['data']) if 'data' in example_input else 'undefined'};"
```
You can see the parenthesis were placed incorrectly. If there is no `data` in `example_input`, the first expression can be simplified and evaluates to 
```python
f"const d = {json.dumps('undefined')};"
>> const d = "undefined"
```
and the second one to 
```python
f"const d = {'undefined'};"
>> const d = undefined
```
In the first case, undefined is a string and thus won't be the actual JS `undefined` object. In the second case it will be instantiated correctly.
----------------------------------
>  just wanted to point this out and make sure everything is ok

It's great to do this when one isn't certain about something. It's better to ask than regret not saying it out loud when things eventually fail. 
However, if there is a simple way of testing the problem, it's also good to take a few minutes to investigate it.

Reply:
I see now yes, thanks for the explanation.

I agree I should check it, as soon as I started reading your comment I asked myself "why didn't I just do that already by myself?".

Resolving